### PR TITLE
Add tests for calculating taxes

### PR DESCRIPTION
**Pull request by korycins**
Add tests for the logic introduced in #2 I've separated it to make review easier.

### CODE REVIEW

**Comment by korycins** [comment]
@witoszekdev I've added you as a reviewer directly as you were picked for reviewer #2. So I thought that you will have the biggest context to make a review here also.

File: __tests__/backend/avataxApi.test.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            city: "Wroclaw",
+            country: "PL",
+            line1: "Teczowa 7",
+            line2: "",
+            line3: "",
+            postalCode: "50-601",
+            region: "",
+          },
+        },
+      },
```
Comment (witoszekdev) [inline __tests__/backend/avataxApi.test.ts:148]: could we extract that into a separate fixtures file?

File: __tests__/pages/webhooks/order-created.test.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  dummyFetchTaxesResponse,
+} from "../../utils";
+import Avatax from "ava-typescript";
+
+jest.mock("next/dist/compiled/raw-body/index.js", () => ({
+  __esModule: true,
+  default: jest.fn((_) => ({
+    toString: () => '{"dummy":12}',
+  })),
+}));
```
Comment (witoszekdev) [inline __tests__/pages/webhooks/order-created.test.ts:20]: *please ignore if that's too much work, I'm just leaving it so we can improve that mocking later*

@mmiszy wrote a nice utility function for mocking requests with `bodyParser` disabled: https://github.com/saleor/react-storefront/blob/main/apps/saleor-app-checkout/test-utils.ts#L36

File: __tests__/pages/webhooks/order-created.test.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+      event: "order_created",
+      domain,
+    });
+
+    const orderPayload = dummyOrderCreatedPayload;
+    req.body = orderPayload;
+
+    await toNextHandler(
+      req as unknown as NextApiRequest,
+      res as unknown as NextApiResponse
```
Comment (witoszekdev) [inline __tests__/pages/webhooks/order-created.test.ts:48]: We could do this type assertion inside the `mockRequest` instead and this way we wouldn't have to write it each time we pass `req` and `res` to the handler function

Example: https://github.com/saleor/react-storefront/blob/main/apps/saleor-app-checkout/test-utils.ts#L24

File: __tests__/utils.ts [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+  };
+  if (configuration.domain !== undefined) {
+    req.headers["saleor-domain"] = configuration.domain;
+  }
+  if (configuration.event !== undefined) {
+    req.headers["saleor-event"] = configuration.event;
+  }
+  if (configuration.signature !== undefined) {
+    req.headers["saleor-signature"] = configuration.signature;
+  }
```
Comment (witoszekdev) [inline __tests__/utils.ts:31]:
```suggestion
  if (configuration.domain) {
    req.headers["saleor-domain"] = configuration.domain;
  }
  if (configuration.event) {
    req.headers["saleor-event"] = configuration.event;
  }
  if (configuration.signature) {
    req.headers["saleor-signature"] = configuration.signature;
  }
```

File: __tests__/utils.ts [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+
+  return { req, res };
+};
+export const dummyCalculateTaxesPayloadForCheckout: TaxBaseSubscriptionFragment =
```
Comment (witoszekdev) [inline __tests__/utils.ts:35]: imho those fixtures should be stored in a separate file because they're really long 😅

**Pull request merged**